misc-utils/uuidd.c

```diff
@@ -526,20 +525,21 @@ static void server_loop(const char *socket_path, const char *pidfile_path,
 				num = 1000;
 			if (num * UUID_LEN > (int) (sizeof(reply_buf) - sizeof(num)))
 				num = (sizeof(reply_buf) - sizeof(num)) / UUID_LEN;
+			memcpy(reply_buf, &num, sizeof(num));
 			__uuid_generate_random((unsigned char *) reply_buf +
 					      sizeof(num), &num);
+			reply_len = sizeof(num) + (UUID_LEN * num);
 			if (uuidd_cxt->debug) {
 				fprintf(stderr, P_("Generated %d UUID:\n",
 						   "Generated %d UUIDs:\n", num), num);
-				for (i = 0, cp = reply_buf + sizeof(num);
-				     i < num;
-				     i++, cp += UUID_LEN) {
+
+				cp = reply_buf + sizeof(num)
+				for (i = 0; i < num; i++) {
 					uuid_unparse((unsigned char *)cp, str);
```

Comment:
This is only part of the patch which makes the code more readable. It would be bettrer to remove the rest.